packages/webapp/src/pages/Integration/FlowPage.tsx

```diff
@@ -62,6 +64,73 @@ export default function FlowPage(props: FlowPageProps) {
         }
     }, [error]);
 
+    /*
+     * If the flow is enabled then we need to make sure to assign
+     * the updated ID to the flow object. This covers the scenario of a user
+     * enabled a flow, then disables it
+     */
+    useEffect(() => {
+        if (!endpoints.allFlows) {
+            return;
+        }
+
+        const { syncs, actions } = endpoints.allFlows;
+
+        if (flow?.type === 'sync') {
+            const sync = syncs.find((sync: Flow) => sync.name === flow.name);
+            if (sync) {
+                setFlow({
+                    ...flow,
+                    id: sync.id,
+                    enabled: sync.enabled,
+                    version: sync.version,
+                    upgrade_version: sync.upgrade_version,
+                    last_deployed: sync.last_deployed
+                });
+            }
+        } else {
+            const action = actions.find((action: Flow) => action.name === flow?.name);
+            if (action) {
```

Comment:
same question. Can action be ever null/undefined?

Reply:
Answered https://github.com/NangoHQ/nango/pull/2491#discussion_r1672773562